doc/ref/gappkg.xml

```diff
@@ -173,5 +184,2074 @@ users.
+this is done by starting &GAP; and running
+<P/>
+&nbsp;&nbsp;<C>gap> GapDocManualLab( "<A>package</A>" );</C>
+<P/>
+<Alt Only="LaTeX">\noindent</Alt>
+for each such <A>package</A> whose manual is cross-referred to 
+(this includes the <Q>main</Q> manuals, e.g.&nbsp;those in the 
+<F>doc/ref</F> and <F>doc/tut</F> directories). 
+For packages using <F>gapmacro.tex</F>-produced manuals, 
+<C>manual.lab</C> is generated by running <C>tex manual</C>
+for each package whose manual is cross-referred to.
+In most cases, packages from the &GAP; distribution will already
+have these files since they will be created as a part of building their 
+manuals (see e.g. the last command of the <F>makedocrel.g</F> script
+from the <Package>Example</Package> package)
+and included in their distribution, so you will probably not need to 
+create <C>manual.lab</C> files yourself.
+<P/>
+To produce an HTML version of the manual one needs the Perl 5 program
+<F>convert.pl</F>. Instructions on its usage are contained as comments
+in this file.
+<P/>
+Finally, to ensure the mathematical formulae are rendered as well as they
+can be in the HTML version,  one  should  also  have  the  program  <C>tth</C>
+(&TeX;  to  HTML  converter);  <F>convert.pl</F>  calls  <F>tth</F>  to  translate
+mathmode formulae to HTML (if it's available). The <F>tth</F> program is  easy
+to compile and can be obtained from
+<URL>http://hutchinson.belmont.ma.us/tth/tth-noncom/download.html</URL>.
+<!--
+A prototype of the command to execute is in the file <F>make_doc</F>; 
+note that the HTML manual is produced in files with <C>.htm</C> extensions 
+in a directory <F>htm</F> outside the <F>doc</F> directory. 
+-->
+<P/>
+
+</Section>
+
+
+<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% -->
+<Section Label="An Example of a GAP Package">
+<Heading>An Example of a GAP Package</Heading>
+
+We illustrate the creation of a &GAP; package by an example of a very basic package. 
+<P/>
+Create the following directories in your home directory: 
+<F>.gap</F>, <F>.gap/pkg</F> and <F>.gap/pkg/test</F>.
+Then inside the directory <F>.gap/pkg/test</F> create an empty file 
+<F>init.g</F>, and a file <F>PackageInfo.g</F> with the following contents:
+<P/>
+<Log><![CDATA[
+SetPackageInfo( rec(
+  PackageName := "test",
+  Version := "1.0",
+  PackageDoc := rec(
+      BookName  := "test",
+      SixFile   := "doc/manual.six",
+      Autoload  := true ),
+  Dependencies := rec(
+      GAP       := "4.9",
+      NeededOtherPackages := [ ["GAPDoc", "1.6"] ],
+      SuggestedOtherPackages := [ ] ),
+  AvailabilityTest := ReturnTrue ) );
+]]></Log>
+<P/>
+This file  declares the &GAP; package with name <Q>test</Q> in version 1.0. 
+The package documentation consists of one autoloaded book; the <C>SixFile</C> 
+component is needed by the &GAP; help system. Package dependencies require at
+least &GAP;&nbsp;4.9 and &GAPDoc; package at version at least 1.6, and these
```

Comment:
This is something that bothers me as well. Some packages have GAPDoc as required package, some as suggested, and some does not specify it in dependencies at all. Because GAPDoc is needed by GAP, everything works anyway. 

There are packages that use GAPDoc functions at runtime, e.g. OpenMath to parse XML. That is, I think, is a real dependency. AutoDoc is another one. For these, it should be required. For the rest - hmm?

Reply:
I've wrote that dependencies are made up for the purposes of this example.